unit/util/irep_sharing.cpp

```diff
@@ -0,0 +1,166 @@
+/// Author: Diffblue Ltd.
+
+/// \file Tests that irep sharing works correctly
+
+#include <testing-utils/catch.hpp>
+#include <util/irep.h>
+
+#ifdef SHARING
+
+SCENARIO("irept_sharing", "[!mayfail][core][utils][irept]")
```

Comment:
I don't think this entire scenario should be allowed to fail. In particular, the first one ("An irept created with move_to_sub and a copy") must not, as otherwise there is no sharing at all. Could you please break this up a bit to filter out those that may be ok to fail?

Reply:
Done that and added another test to show that there is sharing quite a lot of the time even when the one you mention above doesn't pass.